qdrant-landing/content/blog/qdrant-relari.md

```diff
@@ -0,0 +1,289 @@
+db = Qdrant.from_documents(
+    documents,
+    embedding=embeddings,
+    url=os.environ['QDRANT_URL'],  # Qdrant Cloud URL
+    api_key=os.environ['QDRANT_API_KEY'],  # Qdrant Cloud API Key
+    collection_name="gitlab_legal_policies",
+)
+print(f"{len(documents)} chunks loaded into Qdrant Cloud vector database.")
+```
+<aside role="status">
+If you're using the GitLab dataset, you should have 66 chunks.
+</aside>
+
+### Start logging results
+
+Once we've uploaded our data, we'll build a function to run different RAG pipelines over the dataset and log the results.
+
+```python
+# Prepare a function to run different RAG pipelines over the dataset and log the results
+def log_retriever_results(retriever, dataset):
+  log = list()
+  for datum in dataset.data:
+      # First compute the result
+      retrieved_docs = retriever.invoke(datum["question"])
+      # Now log the result in Relari format
+      result = DatasetDatum(
+          label=datum["uid"],
+          data={"retrieved_context": [doc.page_content for doc in retrieved_docs]},
+      )
+      log.append(result)
+  return log
+```
+
+This is the power of combining Qdrant and Relari. Instead of having to build multiple applications, slowly upsert, and retrieve data, you can use both to quickly test different parameters and instantly get results. This evaluation system is built for fast, useful iteration.
+
+### Evaluate results
+
+Now that we have our application built, let's explore how to evaluate Top-K.
+
+#### First attempt
+
+Run an experiment with Top-K, meaning results shown to users will vary between 3, 5, 7, and 9.
+
+```python
+k_values = [3, 5, 7, 9]  # Define the different values of top k to experiment
+
+semantic_retrievers = {}
+semantic_logs = {}
+
+# Run the retrievers on the dataset and log retrieved chunks
+for k in k_values:
+    retriever = db.as_retriever(search_type="similarity", search_kwargs={"k": k})
+    log = log_retriever_results(retriever, dataset)
+
+    semantic_retrievers[f"k_{k}"] = retriever
+    semantic_logs[f"k_{k}"] = log
+    print(f"Results on {dataset.name} by Semantic Retriever with k={k} saved!")
+```
+
+#### Send results to Relari
+
+Submit the results to Relari for evaluation. Relari will run Precision/Recall and Rank-Aware metrics on your results. You can benchmark different results in the UI.
+
+```python
+from relari import Metric
+
+for k in k_values:
+    eval_name = f"Semantic Retriever Evaluation k={k}"
+    eval_data = semantic_logs[f"k_{k}"]
+
+    eval_info = client.evaluations.submit(
+        project_id=proj["id"],
+        dataset=dataset_info["id"],
+        name=eval_name,
+        pipeline=[Metric.PrecisionRecallF1, Metric.RankedRetrievalMetrics],
+        data=eval_data,
+    )
+    print(f"{eval_name} submitted!")
+```
+
+In this example, we can see that with our dataset, if we want the recall to be greater than 85%, we should pick a K value of at least 7.
```

Comment:
I don't see the recall values for different values of k. Is that a part of the missing image below?